pkg/agent/qrm-plugins/network/dynamicpolicy/policy.go

```diff
@@ -0,0 +1,366 @@
+			p.started = true
+		}
+
+		p.Unlock()
+	}()
+
+	if p.started {
+		klog.Infof("[NetworkDynamicPolicy.Start] DynamicPolicy is already started")
+		return nil
+	}
+
+	p.stopCh = make(chan struct{})
+
+	go wait.Until(func() {
+		_ = p.emitter.StoreInt64(util.MetricNameHeartBeat, 1, metrics.MetricTypeNameRaw)
+	}, time.Second*30, p.stopCh)
+
+	go wait.Until(p.applyNetClass, 5*time.Second, p.stopCh)
+
+	return nil
+}
+
+// Stop stops this plugin
+func (p *DynamicPolicy) Stop() error {
+	p.Lock()
+	defer func() {
+		p.started = false
+		p.Unlock()
+
+		klog.Infof("[NetworkDynamicPolicy.Stop] DynamicPolicy stopped")
+	}()
+
+	if !p.started {
+		klog.Warningf("[NetworkDynamicPolicy.Stop] DynamicPolicy already stopped")
+		return nil
+	}
+	close(p.stopCh)
+	return nil
+}
+
+// Name returns the name of this plugin
+func (p *DynamicPolicy) Name() string {
+	return p.name
+}
+
+// ResourceName returns resource names managed by this plugin
+func (p *DynamicPolicy) ResourceName() string {
+	return ResourceNameNetwork
+}
+
+// GetTopologyHints returns hints of corresponding resources
+func (p *DynamicPolicy) GetTopologyHints(ctx context.Context, req *pluginapi.ResourceRequest) (*pluginapi.ResourceHintsResponse, error) {
+	return nil, nil
+}
+
+func (p *DynamicPolicy) RemovePod(ctx context.Context, req *pluginapi.RemovePodRequest) (*pluginapi.RemovePodResponse, error) {
+	if req == nil {
+		return nil, fmt.Errorf("RemovePod got nil req")
+	}
+
+	if p.isCgV2Env {
+		err := p.removePod(req.PodUid)
+		if err != nil {
+			klog.ErrorS(err, "[NetworkDynamicPolicy.RemovePod] remove pod failed with error", "podUID", req.PodUid)
+			return nil, err
+		}
+	}
+
+	return &pluginapi.RemovePodResponse{}, nil
+}
+
+// GetResourcesAllocation returns allocation results of corresponding resources
+func (p *DynamicPolicy) GetResourcesAllocation(ctx context.Context, req *pluginapi.GetResourcesAllocationRequest) (*pluginapi.GetResourcesAllocationResponse, error) {
+	return nil, nil
+}
+
+// GetTopologyAwareResources returns allocation results of corresponding resources as topology aware format
+func (p *DynamicPolicy) GetTopologyAwareResources(ctx context.Context, req *pluginapi.GetTopologyAwareResourcesRequest) (*pluginapi.GetTopologyAwareResourcesResponse, error) {
+	return nil, nil
+}
+
+// GetTopologyAwareAllocatableResources returns corresponding allocatable resources as topology aware format
+func (p *DynamicPolicy) GetTopologyAwareAllocatableResources(ctx context.Context, req *pluginapi.GetTopologyAwareAllocatableResourcesRequest) (*pluginapi.GetTopologyAwareAllocatableResourcesResponse, error) {
+	return nil, nil
+}
+
+// GetResourcePluginOptions returns options to be communicated with Resource Manager
+func (p *DynamicPolicy) GetResourcePluginOptions(context.Context, *pluginapi.Empty) (*pluginapi.ResourcePluginOptions, error) {
+	return &pluginapi.ResourcePluginOptions{
+		PreStartRequired:      false,
+		WithTopologyAlignment: false,
+		NeedReconcile:         false,
```

Comment:
NeedReconcile is true or false? is there a chance that we will update cls in pod annotation after it starts?

Reply:
The reconcilation of net class id is not triggered by QRM, but by a goroutine created in the `Start()` func.

Reply:
for now, we won't support dynamically change, will reconsider in the future if needed